tests/scripts/all.sh

```diff
@@ -2610,14 +2610,11 @@ component_test_no_psa_crypto_full_cmake_asan() {
     msg "test: ssl-opt.sh (full minus PSA crypto)"
     tests/ssl-opt.sh
 
-    msg "test: compat.sh default (full minus PSA crypto)"
-    tests/compat.sh
-
-    msg "test: compat.sh NULL (full minus PSA crypto)"
+    msg "test: compat.sh: NULL (full minus PSA crypto)"
```

Comment:
We don't actually need a separate invocation for null ciphersuites, they're still supported in OpenSSL 1.1.1a (`OPENSSL_NEXT`). We just need a change in `compat.sh`: in OpenSSL 1.1.1, `-cipher ALL@SECLEVEL=0` isn't enough to enable null ciphers, we need `-cipher ALL@SECLEVEL=0:NULL@SECLEVEL=0`. Or maybe `ALL@SECLEVEL=0:COMPLEMENTOFALL@SECLEVEL=0` ? Or is `ALL:COMPLEMENTOFALL` enough (IIRC no: from 4bdb9fbfa2 it's necessary to add `@SECLEVEL` because otherwise we're enabling all ciphers but not all cipher-version combinations)?

Reply:
In the commit message I was precisely saying that it doesn't look like we should need a separate invocation for NULL ciphersuites, but it wasn't working for me when OpenSSL is the server.

Thanks for providing details, I wasn't aware of `@SECLEVEL=0`. From looking at the code with this in mind,, server-side we're using `-cipher` twice on the server command line: once with `ALL,COMPLEMENTOFALL` unconditionally, when later for versions that are not 0.x or 1.0.x, `ALL@SECLEVEL=0`. Not sure how repeated `-cipher` options is supposed to behave, but I think ideally we should only set it once.

(Also, according to the documentation, NULL is not part of ALL, so I guess what we want is `ALL@SECLEVEL=0:COMPLEMENTOFALL@SECLEVEL=0`.)